tendermint-rs/src/validator.rs

```diff
@@ -1,11 +1,40 @@
 //! Tendermint validators
 
-use crate::{account, vote, PublicKey};
+use crate::{account, vote, PublicKey, merkle};
 #[cfg(feature = "serde")]
 use serde::{de::Error as _, Deserialize, Deserializer, Serialize, Serializer};
 #[cfg(feature = "rpc")]
 use subtle_encoding::base64;
 
+
+/// Validator set contains a vector of validators
+#[derive(Debug)]
+pub struct Set {
+    validators: Vec<Info>
+}
+
+impl Set {
+    /// Create a new validator set.
+    /// vals is mutable so it can be sorted by address.
+    pub fn new(mut vals: Vec<Info>) -> Set {
+        vals.sort_by(|v1, v2| v1.address.partial_cmp(&v2.address).unwrap());
+        Set {
+            validators: vals
+        }
+    }
+
+    /// Compute the Merkle root of the validator set
+    pub fn hash(self) -> merkle::Hash {
+        // We need to get from Vec<Info> to &[&[u8]] so we can call simple_hash_from_byte_slices.
+        // This looks like: Vec<Info> -> Vec<Vec<u8>> -> Vec<&[u8]> -> &[&[u8]]
+        // Can we simplify this? 
+        // Perhaps simple_hash_from_byteslices should take Vec<Vec<u8>> directly ? 
```

Comment:
I couldn't go from `Vec<Info>` to `Vec<&[u8]>` directly by eg. `self.validators.into_iter().map(|x| x.hash_bytes().as_slice()` because the hash_bytes retruns an owned value that would then go out of scope while we're trying to keep a reference to it. 

Probably simple_hash_from_byteslices should take `Vec<Vec<u8>>` or at least `&[Vec<u8>]` since we won't need the list of hash_bytes once the Merkle root is computed so it's fine to move them into simple_hash_from_byteslices